File: Lecture11-Close/main.cpp
#include <opencv4/opencv2/opencv.hpp>

using namespace cv;

void close()
{
    Mat img = imread("/home/berkan/CLionProjects/OpenCV/Ders11_Close/word.jpg", IMREAD_GRAYSCALE);
    Mat filter, dilateImg, close;
    filter = Mat::ones(Size(3, 3), img.type());

    namedWindow("Close");
    int min = 0;
    createTrackbar("Close", "Close", &min, 5);

    while (true)
    {
        dilate(img, dilateImg, filter, Point(-1, -1), min, 0, Scalar(255, 255, 255));
        erode(dilateImg, close, filter, Point(-1, -1), min, 0, Scalar(255, 255, 255));
        imshow("Close", dilateImg);

        if (waitKey(10) == 27)
        {
            break;
        }
    }
    destroyWindow("Close");
}

int main() {
    close();
    return 0;
}

File: Lecture11-Close/main.py
import cv2
import numpy as np

def nothing():
    pass

def close():
    img = cv2.imread("cumle.jpg", cv2.IMREAD_GRAYSCALE)
    cv2.namedWindow("Close Image", cv2.WINDOW_AUTOSIZE)
    cv2.threshold(img, 128, 255, cv2.THRESH_BINARY)
    filtre = np.ones((3,3))
    cv2.createTrackbar("Close", "Close Image", 0, 15, nothing)

    while True:
        iteration = cv2.getTrackbarPos("Close", "Close Image")

        erodeImg = cv2.erode(cv2.dilate(img, filtre, dst=(-1,-1), iterations=iteration), filtre, dst=(-1,-1), iterations=iteration)

        cv2.imshow("Close Image", erodeImg)

        if cv2.waitKey(10) == 27:
            break

    cv2.destroyAllWindows()


if __name__ == '__main__':
    close()